### PR TITLE
Feature/add online field

### PR DESCRIPTION
**Pull request by nimzco**
image

cc @prigara @cgrail

### CODE REVIEW

File: src/components/ConferenceNewPage/ConferenceNewPage.tsx [hunk 1]
```diff
@@ -30,6 +31,7 @@ const defaultConference: Conference = {
   cfpUrl: '',
   cfpEndDate: null,
   cocUrl: '',
+  online: false,
```
Comment (nimzco) [inline src/components/ConferenceNewPage/ConferenceNewPage.tsx:34]: Change here.

File: src/components/ConferenceNewPage/ConferenceNewPage.tsx [hunk 2]
```diff
@@ -241,6 +243,7 @@ export default class ConferenceNewPage extends Component<Props> {
         twitter,
         comment,
         cocUrl,
+        online,
```
Comment (nimzco) [inline src/components/ConferenceNewPage/ConferenceNewPage.tsx:246]: Change here.

File: src/components/ConferenceNewPage/ConferenceNewPage.tsx [hunk 3]
```diff
@@ -527,17 +560,18 @@ function getConferenceData(conference: Conference) {
...
+  country: string
+  startDate: Moment | null | undefined
+  endDate: Moment | null | undefined
+  topic: string
+  cfpUrl: string
+  cfpEndDate: Moment | null | undefined
+  twitter: string
+  comment: string
+  cocUrl: string
+  online: boolean
```
Comment (nimzco) [inline src/components/ConferenceNewPage/ConferenceNewPage.tsx:575]: Change here.

File: src/components/ConferenceNewPage/ConferenceNewPage.tsx [hunk 4]
```diff
@@ -249,206 +252,236 @@ export default class ConferenceNewPage extends Component<Props> {
...
+                  id='country'
+                  name='country'
+                  placeholder='Online for online events'
+                  value={country}
+                  onChange={this.handleFieldChange}
+                />
+                {this.errorFor('country', 'Country is required.')}
+              </div>
+            </InputGroup>
+            <InputGroup inline>
```
Comment (nimzco) [inline src/components/ConferenceNewPage/ConferenceNewPage.tsx:367]: Change here.

**Review by nimzco — COMMENTED** [review]
File has been linted. See comments for changes.

**Comment by prigara** [comment]
If it's an online event, I need to add `Online` as city and country AND check the checkbox, right?

**Comment by cgrail** [comment]
I would offer a comobox where you can select: ```onsite, online or hybrid```. If it's pure ```online``` then you don't need to enter a ```city``` or ```country```

**Comment by nimzco** [comment]
Hmm, a combobox might be clearer indeed, let me try that out.
Or, I can put the checbkox before, and if checked, the placeholder in City / Country become "optional"?
I'll play around and keep you posted. ☺️

**Comment by nimzco** [comment]
What do you two think about this? It's a bit simpler than having a dropdown?

Online-checkbox

**Comment by cgrail** [comment]
City and Country should not be optional for online conferences they are not allowed for online conferences. For hybrid conferences it’s allowed but must be a correct city and Country. Otherwise we’ll get again lot of invalid entries with city „online“ and country „Germany“ which we have to manually fix. It would be nice if we don’t have to touch so many PRs and fix them. 
A check upfront to show an error already upfront when somebody enters online just for one field would be also great. 
Currently over 90% of the conferences are online and I’ll have to fix 80% of the PRs because they are wrongly entered.

**Comment by prigara** [comment]
I really like the idea of having a dropdown with options: `Onsite`, `Online` and `Hybrid`.
The `Online steaming` checkbox might also be a bit confusing because even in the past some conferences had streaming but they didn't consider themselves hybrid events.

**Comment by nimzco** [comment]
I'll try that and share a gif asap!

For the naming though, I find hybrid maybe not clear enough? What about something along the lines of:
**Location**
- `Online`
- `In person`
- `In person & online`

Since most of the confs are still online currently, we would default to `Online only` for now and switch it back to `In person only` once the pandemic is over. (If it ever ends 🙊 )

**Comment by nimzco** [comment]
One more question: do you see a hybrid conference as:
```json
{
  "online": true,
  "city": "Paris",
  "country": "France"
}
```

or

```json
{
  "locationType": "hybrid",
  "city": "Paris",
  "country": "France"
}
```
where `locationType` can be `"online" | "hybrid" | "inPerson"`.

I would go for `"online": true` personally, but I'm open to suggestions.

**Comment by cgrail** [comment]
I would also go for "online": true 

🙏 a lot @nimzco

**Comment by nimzco** [comment]
LocationType

**Comment by cgrail** [comment]
Amazing. Thanks a lot.

**Comment by nimzco** [comment]
Awesome, I'll ship that and update the json files ;)

**Pull request merged**